### PR TITLE
fix: metavariable-type correctly matches non-primitive types in php

### LINKED ISSUE #8781: metavariable-type can't match non-primitive types in php
**To Reproduce**
rule:
```
rules:
  - id: test
    patterns:
      - pattern: $X->exec(...)
      - metavariable-type:
          metavariable: $X
          type: BAR
    message: foo bar
    languages:
      - php
    severity: ERROR
```
code:
```
<?php

$foo = new BAR($x, $y, $z);
$foo->exec($args);
```

**Expected behavior**
the last line of the sample code is matched with the provided rule.

### PR DESCRIPTION
**Pull request by ihji**
close #8781

### CODE REVIEW

**Review by aryx — APPROVED** [review]
Very nice!

File: src/parsing/Parse_metavariable_type.ml [hunk 1]
```diff
@@ -37,7 +38,17 @@ let unwrap_type_expr lang expr =
       Some t
   | Lang.Kotlin, G.E { e = G.Cast (t, _, _); _ } -> Some t
   | Lang.Scala, G.E { e = OtherExpr (_, _ :: T t :: _); _ } -> Some t
-  | Lang.Php, G.E { e = G.Cast (t, _, _); _ } -> Some t
+  | ( Lang.Php,
```
Comment (amchiclet) [inline src/parsing/Parse_metavariable_type.ml:41]: I'm not too familiar with this code, but are we missing anything by fully replacing the `G.Cast` case? Or does this change fully subsume it?

Comment (ihji) [reply src/parsing/Parse_metavariable_type.ml:41]: This synthetic expression is solely created to extract the type expression from the provided string, so no we haven't lost anything.

**Comment by ihji** [comment]
PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

**Pull request merged**